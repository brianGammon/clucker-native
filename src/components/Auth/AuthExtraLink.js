/* @flow */
import React from 'react';
import { View, Text, Button } from 'native-base';
import styles from './styles';

type Props = {
  message: string,
  buttonText: string,
  handlePress: () => void,
};

const AuthExtraLink = ({ message, buttonText, handlePress }: Props) => (
  <View style={styles.extraLinks}>
    <Text style={styles.bodyText}>{message}</Text>
    <Button transparent style={{ alignSelf: 'center' }} onPress={handlePress}>
      <Text>{buttonText}</Text>
    </Button>
  </View>
);

export default AuthExtraLink;
