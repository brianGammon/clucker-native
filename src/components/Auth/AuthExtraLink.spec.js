import React from 'react';
import AuthExtraLink from './AuthExtraLink';

describe('AuthExtraLink', () => {
  test('renders', () => {
    const wrapper = shallow(
      <AuthExtraLink
        message="test message"
        buttonText="some text"
        handlePress={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
