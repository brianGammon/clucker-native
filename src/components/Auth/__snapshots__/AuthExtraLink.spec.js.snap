// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthExtraLink renders 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Styled(Text)
    style={
      Object {
        "color": "grey",
      }
    }
  >
    test message
  </Styled(Text)>
  <Styled(Button)
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "center",
      }
    }
    transparent={true}
  >
    <Styled(Text)>
      some text
    </Styled(Text)>
  </Styled(Button)>
</Styled(ViewNB)>
`;
